### DISCUSSION TITLE
Nonlinear transform meta-issue

### DISCUSSION THREAD

**Discussion by asinghvi17** [JuliaPlots/Makie.jl#4166] [Ideas]
# User experience
- [x] Create a wrapper type to allow arbitrary transformations, which act on all coordinates.
- [ ] Document `apply_transform`, and how to overload it ([existing methods are here](https://github.com/JuliaPlots/AbstractPlotting.jl/blob/master/src/layouting/transformation.jl#L210-L245))
- [ ] Create wrapper trait types (`TransformFunc{2D}`, `TransformFunc{NeedsXY}`, etc)
# Low level features
- [x] Make heatmap, surface, mesh, volume, etc work with nonlinear transforms
# Axis integration
- [ ] What kind of axis do we want to use?  Can we specialize on the type of transform?  (e.g. NTuple based transforms should get axes which are plotted using `linesegments`, but nonlinear transforms which transform x and y at the same time should get a densely sampled axis which looks "smooth").  Plus, there should be a way to create a specialized axis for a given transform.

**Comment by ffreyer** [comment]
Is this still useful or can we close this?

**Comment by asinghvi17** [comment]
Probably still useful though I need to update it.  Ideally transforms just subscribe to a trait like interface so we don't need to wrap everything in Makie.PointTrans...
